internal/domain/pull_request.go

```diff
@@ -0,0 +1,30 @@
+package domain
+
+type PullRequestEvent struct {
+	Action      string      `json:"action"`
+	PullRequest PullRequest `json:"pull_request"`
```

Comment:
Consider adding validation methods for the structs to ensure that required fields are present and have valid values. For example, it may not be sufficient to only rely on JSON unmarshalling to check that the `PullRequest` field contains a valid user and repository. This can prevent potential nil pointer dereference issues later in the code.